web/main.ts

```diff
@@ -2490,11 +2523,23 @@ class GitGraphView {
 					});
 					const commitDetails = expandedCommit.commitDetails!;
 					const parents = commitDetails.parents.length > 0
-						? commitDetails.parents.map((parent) => {
+						? commitDetails.parents.map((parent, parentIndex) => {
 							const escapedParent = escapeHtml(parent);
-							return typeof this.commitLookup[parent] === 'number'
-								? '<span class="' + CLASS_INTERNAL_URL + '" data-type="commit" data-value="' + escapedParent + '" tabindex="-1">' + escapedParent + '</span>'
-								: escapedParent;
+
+							const parentWasLoaded = loadedParents.includes(parent);
+							const isComparedToParent = parentIndex + 1 === expandedCommit.parentIndex;
+
+							let parentHtml = abbrevCommit(escapedParent);
+							if(parentWasLoaded) {

```

Comment:
Please implement the changes we discussed previously in this PR:

> The existing implementation only adds internal links to parent commits that are loaded in the view. For this approach to be used, I think we'd have to make it so that all parents are linked, and show an error message if the user tries to follow the link when the parent can't be found in the loaded commits.

> In the context my alternative approach, and how it would impact the internal links, it's possibly still a better solution than the existing behaviour, as at least the user would be shown an error message explaining the situation (e.g. "The commit could not be found in the loaded commits, either: load more commits, or check any filtering that has been applied"), instead of wondering why one parent doesn't have a link, when almost all of them do. The error dialog could even provide an action to "Load More Commits".